contracts/src/systems/transport/contracts/road_systems.cairo

```diff
@@ -0,0 +1,55 @@
+#[system]
+mod road_systems {
+    use eternum::models::position::{Coord};
+    use eternum::models::resources::Resource;
+    use eternum::models::road::{Road, RoadImpl};
+    use eternum::models::owner::Owner;
+    use eternum::models::config::RoadConfig;
+
+    use eternum::constants::ROAD_CONFIG_ID;
+
+    use eternum::systems::transport::interface::road_systems_interface::{
+        IRoadSystems
+    };
+
+
+    #[external(v0)]
+    impl RoadSystemsImpl of IRoadSystems<ContractState> {
+
+        fn create(
+            self:@ContractState, world: IWorldDispatcher, 
+            entity_id: u128, start_coord: Coord, end_coord: Coord, usage_count: usize
+        ) {
+
+            // assert that entity is owned by caller
+            let entity_owner = get!(world, entity_id, Owner);
+            assert(entity_owner.address == starknet::get_caller_address(), 'entity id not owned by caller');
```

Comment:
For the future - we could turn this function into a new Cairo Component and share across contracts